src/subtitle/subtitleTrack.ts

```diff
@@ -0,0 +1,60 @@
+import { MakeRequired } from '../utils';
+
+/**
+ * Supported subtitle/caption file formats.
+ */
+export enum SubtitleFormat {
+  CEA = 'cea',
+  TTML = 'ttml',
+  VTT = 'vtt',
+}
+
+/**
+ * Represents a custom subtitle track source that can be added to `SourceConfig.subtitleTracks`.
+ */
+export interface SubtitleTrack {
+  /**
+   * The URL to the timed file, e.g. WebVTT file.
+   */
+  url?: string;
+  /**
+   * The label for this track.
+   */
+  label?: string;
+  /**
+   * The unique identifier for this track. If no value is provided, a random UUIDv4 will be generated for it.
+   */
+  identifier?: string;
+  /**
+   * Specifies the file format to be used by this track.
+   */
+  format?: SubtitleFormat;
```

Comment:
yes, we have this on iOS, see: https://cdn.bitmovin.com/player/ios/3/docs/Enums/SubtitleFormat.html

Reply:
I see. I think we need to discuss how the API should look on RN

Reply:
Unfortunately, there wasn't much information on the format types supported by Android, so I just assumed it would at least be the same as iOS.

@matamegger Do you have any information on what mime types are supported by Android?

Reply:
Also, @matamegger how do you plan to discuss this? Because I think it'd be best if it were on a github issue.

This way all important decisions made to the project from now on stay registered and publicly visible to anyone willing to support/use the project.

And it could even help with automatic generation of changelogs via CI in the future.

Reply:
Ok, it makes sense. About the API discussion, it surely should be done elsewhere and we can maybe schedule a meeting for it later.

And regarding the Android mime-types support, I'll have to leave it as it is for now i.e. handling only the formats supported by iOS. Otherwise, we'd be creating a lot of surface for miss configurations just like you said. But the API should surely have an escape hatch for this kind of situation as it'll possibly happen again once we start targeting other platforms, like web, tvOS, and so on.

Reply:
Just for the record, `format` has been made required in JS for compatibility with Android's `mimeType` option. See commit: f0b320c3a06dd85fd5386ba44ce5ad0b02f0331d.

Reply:
^ fyi: @rolandkakonyi

Reply:
Nice, making `format` required is no problem for iOS. On iOS there is an API for not providing format only for legacy reasons. It is advised to provide it always.